applications/FSIApplication/tests/FSI_problem_emulator_test.py

```diff
@@ -255,10 +401,63 @@ def ComputeDirichletNeumannResidual(self):
 
         return disp_residual
 
+    def __ComputeInitialResidual(self):
+        # Solve the fluid emulator problem with the initial displacement guess
+        self.__SolveFluidEmulator()
+
+        # Transfer fluid load
+        for i in range(len(self.corrected_iteration_value_f)):
+            self.corrected_iteration_value_f[i] = self.iteration_value_f[i]
+        self.__TransferFluidEmulatorLoad()
+
+        # Solve structure problem
+        self.structure_solver.SolveSolutionStep()
+
+        # Compute the displacement residual
+        disp_residual = self.__ComputeStructureResidual()
+
+        return disp_residual
+
+    def __SolveFluidEmulator(self):
+        # Calculate the spring reactions that emulate the fluid load over the structure interface
+        for i in range(self.GetInterfaceProblemSize()):
+            self.iteration_value_f[2*i] = self.fluid_emulator_stiffness * self.corrected_iteration_value_u[2*i]
+            self.iteration_value_f[2*i+1] = self.fluid_emulator_stiffness * self.corrected_iteration_value_u[2*i+1]
+
+    def __TransferFluidEmulatorLoad(self):
+        interface_submodelpart = self.structure_solver.main_model_part.GetSubModelPart("StructureInterface2D_Solid_interface")
+
+        i = 0
+        for node in interface_submodelpart.Nodes:
+            point_load = Vector(3)
+            point_load[0] = - self.corrected_iteration_value_f[i]
+            point_load[1] = - self.corrected_iteration_value_f[i+1]
+            point_load[2] = 0.0
+            node.SetSolutionStepValue(KratosStructural.POINT_LOAD, 0, point_load)
+            i += 2
+
+    def __ComputeStructureResidual(self):
+        interface_submodelpart = self.structure_solver.main_model_part.GetSubModelPart("StructureInterface2D_Solid_interface")
+
+        # Calculate the structure displacement error
+        disp_residual = Vector(2 * self.GetInterfaceProblemSize())
+
+        i = 0
+        for node in interface_submodelpart.Nodes:
+            disp = node.GetSolutionStepValue(DISPLACEMENT,0)
+            self.iteration_value_u[i] = disp[0]
+            self.iteration_value_u[i+1] = disp[1]
+            disp_residual[i] = disp[0] - self.corrected_iteration_value_u[i]
+            disp_residual[i+1] = disp[1] - self.corrected_iteration_value_u[i+1]
+            i+=2
+
+        return disp_residual
+
 if __name__ == '__main__':
     test = FSIProblemEmulatorTest()
     test.setUp()
     # test.testFSIProblemEmulatorWithAitken()
     # test.testFSIProblemEmulatorWithMVQN()
-    test.testFSIProblemEmulatorWithMVQNRecursive()
+    test.testFSIProblemEmulatorWithIBNMVQN()
+    # test.testFSIProblemEmulatorWithMVQNRecursive()
     test.tearDown()
```

Comment:
you know that there is a python way to do this?
in this case use `python3 FSI_problem_emulator_test.py FSIProblemEmulatorTest.testFSIProblemEmulatorWithIBNMVQN`

see the [python docs](https://docs.python.org/3/library/unittest.html#command-line-interface)

I use it all the time, this is super super useful

Reply:
:open_mouth: I wasn't aware. Thanks for the hint.